### ISSUE TITLE
SC2154 on wait -p

### ISSUE THREAD

**Issue by jkt628** [koalaman/shellcheck#2179]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or "online"): 0.7.1 from linuxbrew
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
wait -n -p reaped && unset "hammers[$reaped]"

```

#### Here's what shellcheck currently says:
SC2154: reaped is referenced but not assigned.

#### Here's what I wanted or expected to see:
nothing.  `wait -p` assigns.

**Comment by audeoudh** [comment]
`read -r var; echo "$var"` is the same scenario (`read` assigns `$var`), and is supported (it does not raises SC2154).

While using an uppercase variable, we do not have SC2154 because of `check-unassigned-uppercase` not enabled by default.  With `-o check-unassigned-uppercase`, we do have it.

**Comment by koalaman** [comment]
Straight forward bug and now fixed. Thanks!
